src/transaction_queue.rs

```diff
@@ -1,35 +1,107 @@
-use std::cmp;
 use std::collections::{HashSet, VecDeque};
+use std::{cmp, fmt};
 
-use rand;
+use rand::{self, Isaac64Rng, Rng};
 
 use traits::Contribution;
+use util::SubRng;
+
+/// An interface to the transaction queue. A transaction queue is a structural part of
+/// `QueueingHoneyBadger` that manages enqueueing of transactions for a future batch and dequeueing
+/// of transactions to become part of a current batch.
+pub trait TransactionQueue<T> {
+    fn is_empty(&self) -> bool;
+    fn push_back(&mut self, t: T);
+    fn choose(&mut self, amount: usize, batch_size: usize) -> Vec<T>;
+    fn remove_all<'a, I>(&mut self, txs: I)
```

Comment:
I think the method docs should be here in the trait, not below in the impl. (Or both?)

Reply:
I chose both to be on the safe side.